[css-anchor-position-1] Disregard transforms when calculating `anchor()` location

RWDavid opened the pull request:
#### df419b7ac5fae718ea1dce64c347a87c9ca8e849
<pre>
[css-anchor-position-1] Disregard transforms when calculating `anchor()` location
<a href="https://bugs.webkit.org/show_bug.cgi?id=277935">https://bugs.webkit.org/show_bug.cgi?id=277935</a>
<a href="https://rdar.apple.com/133652587">rdar://133652587</a>

Reviewed by Alan Baradlay.

CSS transforms should not impact anchor positioning. We should not use
`localToContainerPoint()` to compute the location of an anchor.

This patch instead uses `offsetFromAncestorContainer()`, which ignores
transforms. Some test progressions occur as a result of subtle
differences between how the two functions handle multicol content.

This patch also addresses the fact that an anchor&apos;s enclosing fragmented
flow may actually be an ancestor of the anchor&apos;s containing block.
(Previously it was assumed that the anchor&apos;s enclosing fragmented flow
is always a descendant of the anchor&apos;s containing block.)

* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/anchor-position-multicol-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/anchor-position-multicol-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/anchor-position-multicol-004-expected.txt:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
(WebCore::Style::offsetFromAncestorContainer):
(WebCore::Style::computeAnchorRectRelativeToContainingBlock):

Canonical link: <a href="https://commits.webkit.org/282335@main">https://commits.webkit.org/282335@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a35389f99772b49fe6139be4dbdda29703a4f7b4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62782 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42138 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15378 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66802 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13386 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64902 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49825 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13670 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50641 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9240 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65851 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39184 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54390 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31319 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35892 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11720 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12262 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57438 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12051 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68497 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6728 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11697 "Found 1 new test failure: imported/w3c/web-platform-tests/media-source/URL-createObjectURL-null.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57959 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6760 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54443 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58148 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13941 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5619 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/37958 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39038 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40149 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38780 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/style/AnchorPositionEvaluator.cpp:
```diff
@@ -210,23 +235,26 @@ static LayoutRect computeAnchorRectRelativeToContainingBlock(CheckedRef<const Re
 
-        // Compute the final location.
+        // Change the location to be relative to the anchor's containing block.
+        if (isAncestor(*fragmentedFlow, containingBlock))
```
maybe use` fragmentedFlow->isDescendantOf(containingBlock)`

RWDavid committed:
[css-anchor-position-1] Disregard transforms when calculating `anchor()` location
https://bugs.webkit.org/show_bug.cgi?id=277935
rdar://133652587

Reviewed by Alan Baradlay.

CSS transforms should not impact anchor positioning. We should not use
`localToContainerPoint()` to compute the location of an anchor.

This patch instead uses `offsetFromAncestorContainer()`, which ignores
transforms. Some test progressions occur as a result of subtle
differences between how the two functions handle multicol content.

This patch also addresses the fact that an anchor's enclosing fragmented
flow may actually be an ancestor of the anchor's containing block.
(Previously it was assumed that the anchor's enclosing fragmented flow
is always a descendant of the anchor's containing block.)

* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/anchor-position-multicol-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/anchor-position-multicol-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/anchor-position-multicol-004-expected.txt:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
(WebCore::Style::offsetFromAncestorContainer):
(WebCore::Style::computeAnchorRectRelativeToContainingBlock):

Canonical link: https://commits.webkit.org/282335@main
